pkg/controller/clusterpool/clusterpool_controller_test.go

```diff
@@ -92,6 +108,8 @@ func TestReconcileClusterPool(t *testing.T) {
 		existing                           []runtime.Object
 		noClusterImageSet                  bool
 		noCredsSecret                      bool
+		noCustomization                    bool
```

Comment:
Currently not used in test configs. Are you intending to use it to show what happens when the pool has an inventory that references a CDC that doesn't exist?

Reply:
I'm a bit lost with the testing so I left it to after we finish this review. There are many tests that I can think of but it will take a segnificant amount of time to implement. What are your expectations?

Reply:
> testing so I left it to after we finish this review

Understood and agreed.

> What are your expectations?

We start with the expectation that unit tests will cover 100% of the code touched by the PR. Once CI is passing, the `coverage` job will drop a comment with links to a report that makes it easy to see what's covered and what's not.

There are certainly some code paths that are not feasible to cover -- e.g. we don't tend to worry about "AddToManager" and other framework-y stuff; and it's tough to demonstrate "coverage" of type definitions and such.

One step up from that is simulating errors from external calls. We have mocks for a lot of things, like the k8s client, that let you a) assert that those methods were invoked with the parameters you expect; and b) force that call to return something specific to make your test run behave a certain way. [Example](https://github.com/openshift/hive/blob/aa9b5541222e043caae5a95bd8660ce2f31d7340/pkg/controller/clustersync/clustersync_controller_test.go#L364). We probably wouldn't hold up the PR if you skipped _some_ of those, but it would be nice to cover them if it's not too much trouble.

And the last tier is covering the actual code you wrote/touched. There _shouldn't_ be a reason you can't cover every line of that.

You'll find once you've addressed one or two test cases of each type, adding subsequent ones becomes a lot easier. We're happy to help get you started if you're having trouble.

But yes, we can worry about that once we've gotten closer to what the code side is going to look like in the end.